src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -8518,6 +8575,15 @@ void Assembler::vpmuludq(XMMRegister dst, XMMRegister nds, XMMRegister src, int
   emit_int16((unsigned char)0xF4, (0xC0 | encode));
 }
 
+void Assembler::vpmuldq(XMMRegister dst, XMMRegister nds, XMMRegister src, int vector_len) {
+  assert(vector_len == AVX_128bit ? VM_Version::supports_avx() :
+        (vector_len == AVX_256bit ? VM_Version::supports_avx2() : VM_Version::supports_evex()), "");
+  // TODO check what legacy_mode needs to be set to
```

Comment:
Drive-by comment: There's a TODO left in here.

Reply:
Actually, I was hoping that I would learn that from a reviewer @vpaprotsk or @vnkozlov , do you know? I was not able to figure it out from the manual what it should be.

Reply:
I am looking it up.. gimme a bit? Though I was thinking about the UseSHA3Intrinsic flag.. missed this one..

Reply:
legacy_mode should be false here. This instruction is promotable to evex encoding if higher bank registers (XMM16 and above) are used. It is not a legacy instruction so legacy_mode should be false. Examples of legacy instructions are vptest, vpblend*.

Reply:
Thank you!